lnst/Controller/Recipe.py

```diff
@@ -106,6 +106,8 @@ def __init__(self, **kwargs):
         params = ((name, val) for name, val in attrs.items() if isinstance(val, Param))
         for name, val in params:
             if name in kwargs:
+                if isinstance(val, ConstParam):
+                    raise RecipeError(f"Value for constant parameter {name} was provided")
```

Comment:
imo this exception should be a `ParamError` raised from the `validate` method of the `ConstParam` class and should check if the provided value matches the required constant, otherwise raise the exception.

this would:
1. keep responsibility for ConstParam inside the ConstParam class instead of extending the Recipe class
2. make it easier to use in automation where we simply expand `**params` dynamically to generate the recipe classes - we wouldn't have to implement a `if "param" in params: del params["param"]` for situations where the values are actually ok.

Reply:
1. sure, makes sense, but
2. I thought you wanted to raise an exception if a `ConstParam` argument was provided?

Reply:
i think this is fine:

```python
class MyRecipe():
    p = ConstParam(value=123)

MyRecipe(p=123) # this is ok
MyRecipe(p=321) # this raises an exception that the constant is being changed
```

Reply:
Fixed, please resolve.